src/main/java/mc/spigot/weatherbylocation/ServerLocator.java

```diff
@@ -70,6 +84,32 @@ private static LocationData geolocateIpAddress(String ipAddress) throws IOExcept
         locationResult.city = jsonNode.get("city").asText();
         locationResult.latitude = jsonNode.get("lat").asDouble();
         locationResult.longitude = jsonNode.get("lon").asDouble();
+
+        // Save new locationData object to config.yml file
+        config.set("ipAddress", ipAddress);
```

Comment:
Let's break out these `config.set()` calls into their own function. Maybe called something like `saveDataToConfig()`?

Then call the function here and pass in `ipAddress` and `locationResult`.

Reply:
Understood, the block that sets values to the config file are to be replaced by a function call.

Since this is inside the geolocateIpAddress() function that passes "FileConfiguration config" as one of its parameters is it safe to assume that either the saveDataToConfig() method should be overloaded to take a 3rd parameter (config) in order to avoid making multiple calls to this.plugin.getConfig(); or that it should just always have at least the 3 parameters ? (ipAddress, locationResult, config)

Reply:
I think it just makes sense to call `plugin.getConfig()` in the function. I don't think there's really much cost to calling it multiple times.

Reply:
Ok duly noted, will revert the recent commit and adjust

thanks

Reply:
Upon giving it another look, I am not sure if geolocateIpAddress() should still be a static method ? 

If it is to make a call to the function saveDataToConfig() we are presented with the following issues :  

- if saveDataToConfig() is non-static then it will need to make a call to get instance of config with this.plugins.getConfig() , but geolocateIpAddress() cannot make a call to the instance-specific function as it is static 

- if saveDataToConfig() is static then we would not be using class instances, and so the most straightforward way to get access to the config.yml file would be to pass it as a parameter - which seems to not be something desired as of now